cmd/snap/cmd_run.go

```diff
@@ -1579,11 +1579,21 @@ func (x *cmdRun) runSnapConfine(info *snap.Info, runner runnable, beforeExec fun
 			if err != cgroup.ErrCannotTrackProcess {
 				return err
 			}
-			// If we cannot track the process then log a debug message.
-			// TODO: if we could, create a warning. Currently this is not possible
-			// because only snapd can create warnings, internally.
-			logger.Debugf("snapd cannot track the started application")
-			logger.Debugf("snap refreshes will not be postponed by this process")
+			switch runner.info.Base {
+			case "", "core", "core18", "core20", "core22":
+				// If we cannot track the process then log a debug message.
+				// TODO: if we could, create a warning. Currently this is not possible
+				// because only snapd can create warnings, internally.
+				logger.Debugf("snapd cannot track the started application")
+				logger.Debugf("snap refreshes will not be postponed by this process")
+			default:
```

Comment:
We need to check if the snap is allowed to manage it own device cgroup, i.e. `self-managed` or marked as `non-strict` as it skips the failing device cgroup setup in snap-confine anyways

https://github.com/canonical/snapd/blob/254ac100937a4b823fc4ed6bcd3864d2305c7c02/cmd/snap-confine/snap-confine.c#L788-L795

Reply:
yes, that is what I had in mind

Reply:
we don't know if there are snaps out in the wild configured this way which might break by adding this restriction since core base 24 has been out for a while now